sonar-plugin-api/src/main/java/org/sonar/api/batch/Initializer.java

```diff
@@ -20,30 +20,39 @@
 package org.sonar.api.batch;
 
 import org.sonar.api.ExtensionPoint;
+import org.sonar.api.batch.fs.FileSystem;
 import org.sonar.api.resources.Project;
 
 /**
  * <p>
- * Initializer can execute external tool (like a Maven plugin), change project configuration. For example CoberturaMavenInitializer invokes
- * the Codehaus Cobertura Mojo and sets path to Cobertura report according to Maven POM.
- * 
- * 
+ * Initializer are executed at the very beginning of each module analysis, prior the core listing files to be analyzed. It means {@link FileSystem} should not be accessed.
  * <p>
- * Initializers are executed first and once during project analysis.
- * 
- * 
  * @since 2.6
  */
 @BatchSide
 @ExtensionPoint
 public abstract class Initializer implements CheckProject {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "CheckProject"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)